File: 1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums = sorted((v, i) for i, v in enumerate(nums))
        i = 0
        j = len(nums) - 1
        while i < j:
            s = nums[i][0] + nums[j][0]
            if s == target:
                return nums[i][1], nums[j][1]
            elif s > target:
                j -= 1
            else:
                i += 1

File: 1.cpp
#include <algorithm>

struct Solution {
    vector<int> twoSum(const vector<int>& nums, int target) {
        size_t i, j, n = nums.size();
        vector<int> indices;
        indices.resize(n);
        for (i = 0; i < n; ++i){
            indices[i] = i;
        }
        sort(indices.begin(), 
             indices.end(), 
             [&nums](size_t i, size_t j){return nums[i] < nums[j];});
        i = 0;
        j = n - 1;
        int sum;
        while (i < j){
            sum = nums[indices[i]] + nums[indices[j]];
            if (sum == target){
                return {indices[i], indices[j]};
            } else if (sum < target){
                ++i;
            } else {
                --j;
            }
        }
        return {0, 0};
    }
};
